Src/EB/AMReX_EB_chkpt_file.cpp

```diff
@@ -0,0 +1,353 @@
+                                 Array<MultiFab,AMREX_SPACEDIM>& facecent,
+                                 Array<MultiFab,AMREX_SPACEDIM>& edgecent,
+                                 MultiFab& levelset, int ng_gfab, const Geometry& geom,
+                                 const IntVect& ngrow_finest, bool extend_domain_face,
+                                 int max_grid_size) const
+{
+    Real prob_lo[AMREX_SPACEDIM];
+    Real prob_hi[AMREX_SPACEDIM];
+
+    std::string File(m_restart_file + "/Header");
+
+    if (amrex::Verbose()) amrex::Print() << "file=" << File << std::endl;
+
+    VisMF::IO_Buffer io_buffer(VisMF::GetIOBufferSize());
+
+    Vector<char> fileCharPtr;
+    ParallelDescriptor::ReadAndBcastFile(File, fileCharPtr);
+    std::string fileCharPtrString(fileCharPtr.dataPtr());
+    std::istringstream is(fileCharPtrString, std::istringstream::in);
+
+    std::string line, word;
+
+    std::getline(is, line);
+
+    int nlevs;
+    is >> nlevs;
+    gotoNextLine(is);
+    AMREX_ASSERT(nlevs == 1);
+
+    std::getline(is, line);
+    {
+        std::istringstream lis(line);
+        int i = 0;
+        while (lis >> word) {
+            prob_lo[i++] = std::stod(word);
+        }
+    }
+
+    std::getline(is, line);
+    {
+        std::istringstream lis(line);
+        int i = 0;
+        while (lis >> word) {
+            prob_hi[i++] = std::stod(word);
+        }
+    }
+
+    for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(Math::abs(prob_lo[idim] - geom.ProbLo()[idim]) < std::numeric_limits<Real>::epsilon(),
+                                         "EB2::ChkptFile cannot read from a different problem domain");
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(Math::abs(prob_hi[idim] - geom.ProbHi()[idim]) < std::numeric_limits<Real>::epsilon(),
+                                         "EB2::ChkptFile cannot read from a different problem domain");
+    }
+
+    IntVect ngrow_chkptfile;
+    std::getline(is, line);
+    {
+        std::istringstream lis(line);
+        int i = 0;
+        while (lis >> word) {
+            ngrow_chkptfile[i++] = std::stoi(word);
+        }
+    }
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(ngrow_chkptfile == ngrow_finest, "EB2::ChkptFile cannot read from different ngrow");
+
+    bool edf_chkptfile;
+    is >> edf_chkptfile;
+    gotoNextLine(is);
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(extend_domain_face == edf_chkptfile,
+                                     "EB2::ChkptFile cannot read from different extend_domain_face");
+
+    int mgs_chkptfile;
+    is >> mgs_chkptfile;
+    gotoNextLine(is);
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(max_grid_size == mgs_chkptfile,
+                                     "EB2::ChkptFile cannot read from different max_grid_size");
+
+    BoxArray cut_grids_ba, covered_grids_ba;
+
+    if (amrex::Verbose()) amrex::Print() << "Loading cut_grids\n";
+    cut_grids_ba.readFrom(is);
```

Comment:
Also, I don't think you need to have `cut_grids_ba` and `covered_grids_ba`.  You can just read directly.
```
cut_grids.readFrom(is);
```